### PR TITLE
Make typecheck optional in python workflow

### PR DESCRIPTION
**Pull request by f-fl0**

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -32,7 +35,11 @@ jobs:
           python-version: '3.9.5'
           architecture: x64
       - name: Install flake8 and plugins
-        run: pip install --disable-pip-version-check flake8-annotations==2.6.2 flake8-black==0.3.3 flake8-isort==4.1.1
+        run: |
+          pip install --disable-pip-version-check flake8-black==0.3.3 flake8-isort==4.1.1
+          if [ "${{ inputs.run-typecheck }}" = "true" ]; then
+            pip install --disable-pip-version-check flake8-annotations==2.6.2
+          fi
```
Comment (at-wat) [inline .github/workflows/python.yml:42]:
```suggestion
        run: pip install --disable-pip-version-check flake8-black==0.3.3 flake8-isort==4.1.1
      - name: Install flake8-annotations
        if: inputs.run-typecheck
        run: pip install --disable-pip-version-check flake8-annotations==2.6.2
```

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -68,8 +75,9 @@ jobs:
           exit $exit_code
   typecheck:
     runs-on: ubuntu-latest
+    if: ${{ inputs.run-typecheck }}
```
Comment (at-wat) [inline .github/workflows/python.yml:78]:
```suggestion
    if: inputs.run-typecheck
```

**Review by at-wat — COMMENTED** [review]
I'm not very sure why `lint` job needs a change.
If `lint` job currently contains duplicated typecheck, it would be better to remove.

**Comment by at-wat** [comment]
> I'm not very sure why `lint` job needs a change.
> If `lint` job currently contains duplicated typecheck, it would be better to remove.

Does it mean `lint` job checks only type annotation existence and `typecheck` job checks types?

**Comment by f-fl0** [comment]
> > I'm not very sure why `lint` job needs a change.
> > If `lint` job currently contains duplicated typecheck, it would be better to remove.
> 
> Does it mean `lint` job checks only type annotation existence and `typecheck` job checks types?

I think there is some overlap between `flake8-annotations` and `mypy` but according to https://pypi.org/project/flake8-annotations/, `flake8-annotations` is not meant to replace `mypy`. In my opinion, I think it would be better to use only `mypy` to handle everything related to type checking.

**Review by at-wat — APPROVED** [review]

**Pull request merged**